### PR TITLE
Added Shell Sort [Javascript]

### LINKED ISSUE #290: Shell Sort [JavaScript]
#### This issue is about [Shell Sort](https://en.wikipedia.org/wiki/Shellsort).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.

### PR DESCRIPTION
**Pull request by mohitkyadav**
Fixes #290 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by Monal5031** [comment]
@mohitkyadav Add comments to functon about params and return values.

File: README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
 | [Quicksort](https://en.wikipedia.org/wiki/Quicksort) | [:white_check_mark:](quicksort/quicksort.c) | | [:white_check_mark:](quicksort/QuickSort.java) | [:white_check_mark:](quicksort/quick_sort.py) | [:white_check_mark:](quicksort/quick_sort.go) | |
 | [Rod Cutting Problem](http://www.geeksforgeeks.org/dynamic-programming-set-13-cutting-a-rod/) | [:white_check_mark:](rod_cutting_problem/rod_cutting.c) | | [:white_check_mark:](rod_cutting_problem/RodCutting.java) | [:white_check_mark:](rod_cutting_problem/rod_cutting.py) | [:white_check_mark:](rod_cutting_problem/rod_cutting.go) | |
-| [Shell Sort](https://en.wikipedia.org/wiki/Shellsort) | | [:white_check_mark:](shell_sort/ShellSort.cpp) | | [:white_check_mark:](/shell_sort/shell_sort.py) | [:white_check_mark:](shell_sort/shell_sort.go) | |
+| [Shell Sort](https://en.wikipedia.org/wiki/Shellsort) | | [:white_check_mark:](shell_sort/ShellSort.cpp) | | [:white_check_mark:](/shell_sort/shell_sort.py) | [:white_check_mark:](shell_sort/shell_sort.go) | [:white_check_mark:](shell_sort/shell_sort.js) |
```
Comment (aashutoshrathi) [inline README.md]: change file name to `shellSort.js`

Comment (mohitkyadav) [inline README.md]: What is the need?

Comment (aashutoshrathi) [inline README.md]: @aviaryan Sir just made us change our file names to `lowerCamelCase`

Comment (mohitkyadav) [inline README.md]: :+1:  on it

File: shell_sort/shellSort.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Worst case time complexity = O(n^2)
+ * Best case complexity = O(nlog(n))
+ * @param {Array} data
+ * returns sorted data
+ */
+function shellSort (data) {
```
Comment (aashutoshrathi) [inline shell_sort/shellSort.js]: @mohitkyadav Are you working on this ?

Comment (mohitkyadav) [inline shell_sort/shellSort.js]: I don't see any way here to reduce cyclomatic complexity. Do you?

File: shell_sort/shellSort.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	for (let i = Math.floor(data.length / 2); i > 0; i = Math.floor(i / 2)) {
+		for (let j = i; j < data.length; j++) {
+			for (let k = j - i; k >= 0; k -= i) {
+				if (data[k + i] >= data[k]) {
+					break;
+				} else {
+					var temp = data[k];
```
Comment (aviaryan) [inline shell_sort/shellSort.js]: Declaring `var temp` at the top will be more efficient since it will avoid re-allocations. 
Also use `let`.

**Pull request merged**